### ISSUE TITLE
[SUGGESTION] Sub-assembly blueprints

### ISSUE THREAD

**Issue by StarrKiss** [tsunamayo/Starship-EVO#2581]
Currently, there is no such thing as a 'sub-assembly', this is really annoying when, among other things, placing turrets, or placing other objects on rotators which don't mirror. This would also be useful for placing engine pods or modular hull sections and linking them together.

**Comment by EndoSkull** [comment]
I'll include this here because I think it's related - Must have tool:

Resizable Selection Box - Copy / Cut / Paste 
Save to loadable object

I just spent 20 mins make a small hallway and i was like nope not going to keep building all this over again :-)

**Comment by tsunamayo** [comment]
The problem is that you can cram an ungodly amount of detail in a small space in this game. So instead of using normal hull block you would use a super complex piece of custom wall because it looks fancier, but it would have 100x piece instead of 1... It would make the game lag a lot.
Unfortunately the brick are very different from other game voxel...
Maybe after some improvements to the engine maybe...

**Comment by EndoSkull** [comment]
I understand this perspective - as other games have the same issue - players create detail.
That being understood - people will and are creating detail regardless - by not implementing this you're missing a tool in the kit.

Your best bet would be to assign a "complex" rating to a creation that acts as a warning

Builders are going to build so warn them but don't make it difficult to do so.

**Comment by tsunamayo** [comment]
yes but with such a tool that would be super easy to do. Other game dont have the same system as starship EVO, they usually have something without stretch but where inner faces are not drawn, and they cant micro detail.

**Comment by RedVanGhoul** [comment]
Like you said, maybe when the engine gets improved.

Personally I'd like a system like this for macro parts as well.  When making long, tubular ships it would be a lot easier to make a 16 block long segment and then copy/paste it 20 times.
